### PR TITLE
add MatrixFree::get_cell_level_and_index()

### PR DESCRIPTION
**Pull request by jwitte08**
Hi,

this PR adds the functionality of returning the pair (level, index) instead of the `cell_iterator` to `MatrixFree`. Since `cell->level()` and `cell->index()` (with `cell` being a derivative of `TriaIterator`) return integers I have decided to return a `std::pair<int, int>` although the level and index are internally stored as unsigned integers.

### CODE REVIEW

File: include/deal.II/matrix_free/matrix_free.h [hunk 1]
```diff
@@ -1455,6 +1455,18 @@ class MatrixFree : public Subscriptor
                     const unsigned int vector_number,
                     const unsigned int dof_handler_index = 0) const;
 
+  /**
+   * This returns the pair of the cell level and cell index, that is
+   * <tt>cell->level()</tt> and <tt>cell->index()</tt> given a
```
Comment (tjhei) [inline include/deal.II/matrix_free/matrix_free.h]:
```suggestion
   * <tt>cell->level()</tt> and <tt>cell->index()</tt>, given a
```

File: include/deal.II/matrix_free/matrix_free.h [hunk 2]
```diff
@@ -1455,6 +1455,18 @@ class MatrixFree : public Subscriptor
                     const unsigned int vector_number,
                     const unsigned int dof_handler_index = 0) const;
 
+  /**
+   * This returns the pair of the cell level and cell index, that is
+   * <tt>cell->level()</tt> and <tt>cell->index()</tt>, given a
+   * TriaIterator <tt>cell</tt>, to a given cell in the renumbering of
+   * this structure.
+   *
+   * See get_cell_iterator() for more information.
```
Comment (masterleinad) [inline include/deal.II/matrix_free/matrix_free.h]: That sounds pretty convoluted. What about referring to `get_cell_iterator()` similar to
"This returns level and index for the cell that would be returned by get_cell_iterator() when called with the same arguments."?

Comment (jwitte08) [reply include/deal.II/matrix_free/matrix_free.h]: Thanks. Fixed it.

**Comment by masterleinad** [comment]
Would you mind creating changelog entries for your recent contributions? I'll merge this here anyway so we don't have to wait for the tester again.

**Pull request merged**